pkg/types/vsphere/validation/platform.go

```diff
@@ -49,6 +50,25 @@ func ValidatePlatform(p *vsphere.Platform, agentBasedInstallation bool, fldPath
 		if p.Hosts != nil {
 			allErrs = append(allErrs, validateHosts(p, c, fldPath.Child("hosts"))...)
 		}
+	} else {
+		// agent-based installation allows the credentials to be optional
+		if len(p.VCenters) > 0 {
+			if p.VCenters[0].Username != "" && p.VCenters[0].Password != "" &&
+				p.VCenters[0].Server != "" && len(p.VCenters[0].Datacenters) != 0 {
```

Comment:
Maybe I'm missing something, but shouldn't there be an OR between these instead of AND? Otherwise it won't catch if one of them is missing and there's no point calling validateVCenters() which checks each value. It actually seems like this whole check should be removed.

Reply:
Yes, it is a bit confusing here. Let me reword it and change the logic a bit to make it more clear. The goal was to allow credentials to be optional and to not validate if the credentials are not provided. I think this actually belongs in AGENT-717. There is overlapping logic happening here with asset/agent/installconfig.go.